.github/actions/disk-cleanup/action.yaml

```
@@ -0,0 +1,24 @@
+name: Disk cleanup
+description: "Cleanup disk space"
+runs:
+  using: composite
+  steps:
+    - name: Free up disk space
+      shell: bash
+      run: |
+        # https://github.com/actions/runner-images/issues/2840#issuecomment-2272410832
+        sudo rm -rf \
+          "$AGENT_TOOLSDIRECTORY" \
```

Probably worth protecting this against accidentally passing additional arguments to rm. Also not sure where this variable is coming from either, unless that's stock standard GitHub environment stuff I'd feel a little more comfortable not putting it there.
```suggestion
        sudo rm -rf -- \
```

---

good point, the value is pointing to /opt/hostedtoolcache, I think we can just hard coded it.